File: Java/src/test/java/org/codecop/lcdnumbers/SpringBootStartTest.java
package org.codecop.lcdnumbers;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.endsWith;

import java.io.IOException;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.system.CapturedOutput;
import org.springframework.boot.test.system.OutputCaptureExtension;
import org.springframework.test.context.junit.jupiter.SpringExtension;

@ExtendWith(SpringExtension.class)
@SpringBootTest(args = { "1234567890", "2" })
@ExtendWith(OutputCaptureExtension.class)
public class SpringBootStartTest {

    @Test
    public void lcdAllDigitsOfSize2(CapturedOutput output) throws IOException {
        assertThat(output.getOut(), endsWith(Input.join(Input.allDigitsSize2())));
    }
}

File: Kotlin-Spring/src/test/kotlin/org/codecop/lcdnumbers/SpringBootStartTest.kt
package org.codecop.lcdnumbers

import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.system.CapturedOutput
import org.springframework.boot.test.system.OutputCaptureExtension

@SpringBootTest(args = ["1234567890", "2"])
@ExtendWith(OutputCaptureExtension::class)
class SpringBootStartTest {
    @Test
    fun lcdAllDigitsOfSize2(output: CapturedOutput) {
        MatcherAssert.assertThat(output.out, Matchers.endsWith(allDigitsSize2().joinLines()))
    }
}